Fix (most of) the test suite

kmod opened the pull request:
The main issue was with the rename to python3.8-pyston2.2, I backed that out in one place so that we name eggs "py3.8" instead of "py3.8-pyston2.2".

The test suite is now passing except for numpy due to #62

kmod committed:
Improve test canonicalization

- Improve the filename regex to catch some cases it didn't before and to not over-match
- My system python is python 3.8.10 so we need to match more micro digits

kmod committed:
Change _PY_VERSION_SHORT back to 3.8

from 3.8-pyston2.2

Looks like it's documented to just be MAJOR.MINOR, and setuptools hard-codes that

uwsgi uses VERSION so that still works

kmod committed:
Need a couple more libraries to pass the testsuite

kmod committed:
this path needed to be updated in the name change

kmod committed:
Update the version of numpy we test against

I did this in the process of testing #62, and also had to fix
a new issue that cropped up.

undingen commented:
Except the numpy recursion crashes I'm seeing a problem with `pyston/test/test_venvs.py`:
```
While installing:  o = getOutput([pip, "install", "numpy==1.19.4"], env={"PYSTON_UNSAFE_ABI":"1"}).decode("utf8")
pyston/test/test_venvs.py
[...]
    Traceback (most recent call last):
      File "/tmp/tmpnizoc7lf/env/lib/python3.8-pyston2.2/site-packages/pip/_vendor/pep517/_in_process.py", line 257, in <module>
        main()
      File "/tmp/tmpnizoc7lf/env/lib/python3.8-pyston2.2/site-packages/pip/_vendor/pep517/_in_process.py", line 240, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/tmp/tmpnizoc7lf/env/lib/python3.8-pyston2.2/site-packages/pip/_vendor/pep517/_in_process.py", line 110, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
      File "/tmp/pip-build-env-vk5m852m/overlay/lib/python3.8-pyston2.2/site-packages/setuptools/build_meta.py", line 157, in prepare_metadata_for_build_wheel
        self.run_setup()
      File "/tmp/pip-build-env-vk5m852m/overlay/lib/python3.8-pyston2.2/site-packages/setuptools/build_meta.py", line 248, in run_setup
        super(_BuildMetaLegacyBackend,
      File "/tmp/pip-build-env-vk5m852m/overlay/lib/python3.8-pyston2.2/site-packages/setuptools/build_meta.py", line 142, in run_setup
        exec(compile(code, __file__, 'exec'), locals())
      File "setup.py", line 508, in <module>
        setup_package()
      File "setup.py", line 500, in setup_package
        setup(**metadata)
      File "/tmp/pip-install-hnlqae1l/numpy/numpy/distutils/core.py", line 169, in setup
        return old_setup(**new_attr)
      File "/tmp/pip-build-env-vk5m852m/overlay/lib/python3.8-pyston2.2/site-packages/setuptools/__init__.py", line 165, in setup
        return distutils.core.setup(**attrs)
      File "/home/marius/dev/pyston_v2/pyston/build/cpython_unopt_install/usr/lib/python3.8-pyston2.2/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/home/marius/dev/pyston_v2/pyston/build/cpython_unopt_install/usr/lib/python3.8-pyston2.2/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/home/marius/dev/pyston_v2/pyston/build/cpython_unopt_install/usr/lib/python3.8-pyston2.2/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-build-env-vk5m852m/overlay/lib/python3.8-pyston2.2/site-packages/setuptools/command/dist_info.py", line 31, in run
        egg_info.run()
      File "/tmp/pip-install-hnlqae1l/numpy/numpy/distutils/command/egg_info.py", line 24, in run
        self.run_command("build_src")
      File "/home/marius/dev/pyston_v2/pyston/build/cpython_unopt_install/usr/lib/python3.8-pyston2.2/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/home/marius/dev/pyston_v2/pyston/build/cpython_unopt_install/usr/lib/python3.8-pyston2.2/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-install-hnlqae1l/numpy/numpy/distutils/command/build_src.py", line 144, in run
        self.build_sources()
      File "/tmp/pip-install-hnlqae1l/numpy/numpy/distutils/command/build_src.py", line 155, in build_sources
        self.build_library_sources(*libname_info)
      File "/tmp/pip-install-hnlqae1l/numpy/numpy/distutils/command/build_src.py", line 288, in build_library_sources
        sources = self.generate_sources(sources, (lib_name, build_info))
      File "/tmp/pip-install-hnlqae1l/numpy/numpy/distutils/command/build_src.py", line 378, in generate_sources
        source = func(extension, build_dir)
      File "numpy/core/setup.py", line 663, in get_mathlib_info
        raise RuntimeError("Broken toolchain: cannot link a simple C program")
```
Besides this two issues `make test` is working for me again :)

kmod committed:
Pretty sure this duplicate line was a mistake

kmod committed:
Fix this test

1) We removed PYSTON_UNSAFE_ABI so we don't need to test that
2) I didn't realize that passing the env kw resets all the variables

kmod committed:
Update to numpy version which includes the fixes for #62

kmod committed:
Force the numpy testsuite into pyston-compatibility mode

kmod commented:
I fixed that issue (seems like it was broken for a while), and integrated the new numpy fixes, and looks like things are working!
